From 949d4b715cd3df55ebbbff7fb2ed691dc3987007 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Sat, 21 Mar 2020 15:54:25 +0800
Subject: [PATCH] Revert "PM / devfreq: Add new device link for sysfs"

This reverts commit 32948d8b864c2a30cbd68c19c3400db7bdc1fbd4.

See commit a8b1fa6c47bc ("Revert "PM / devfreq: Modify the device name as devfreq(X) for sysfs"")

Conflicts:
	drivers/devfreq/devfreq.c

Change-Id: If4fec73bb1cf38afffd4699dfe25b8c32ea7472f
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/devfreq/devfreq.c | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/drivers/devfreq/devfreq.c b/drivers/devfreq/devfreq.c
index 1e94fae59696..1b51b9150a38 100644
--- a/drivers/devfreq/devfreq.c
+++ b/drivers/devfreq/devfreq.c
@@ -28,9 +28,6 @@
 #include <linux/hrtimer.h>
 #include <linux/of.h>
 #include "governor.h"
-#ifdef CONFIG_ARCH_ROCKCHIP
-#include "../base/base.h"
-#endif
 
 static struct class *devfreq_class;
 
@@ -678,18 +675,6 @@ struct devfreq *devfreq_add_device(struct device *dev,
 		goto err_out;
 	}
 
-#ifdef CONFIG_ARCH_ROCKCHIP
-	if (sysfs_create_link(&devfreq->dev.class->p->subsys.kobj,
-			      &devfreq->dev.kobj, dev_name(dev))) {
-		dev_err(dev, "failed to create devfreq %s link\n",
-			dev_name(dev));
-		device_unregister(&devfreq->dev);
-		mutex_unlock(&devfreq->lock);
-		put_device(&devfreq->dev);
-		goto err_out;
-	}
-#endif
-
 	devfreq->trans_table =
 		devm_kzalloc(&devfreq->dev,
 			     array3_size(sizeof(unsigned int),
@@ -759,10 +744,6 @@ int devfreq_remove_device(struct devfreq *devfreq)
 	if (devfreq->governor)
 		devfreq->governor->event_handler(devfreq,
 						 DEVFREQ_GOV_STOP, NULL);
-#ifdef CONFIG_ARCH_ROCKCHIP
-	sysfs_delete_link(&devfreq->dev.class->p->subsys.kobj,
-			  &devfreq->dev.kobj, dev_name(devfreq->dev.parent));
-#endif
 	device_unregister(&devfreq->dev);
 
 	return 0;
-- 
2.35.3

